src/EditorFeatures/Core/Extensibility/VersionSelector.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Extensibility.Composition;
+
+namespace Microsoft.CodeAnalysis.Editor
+{
+    internal sealed class VersionSelector
+    {
+        public static T Select<T>(IEnumerable<Lazy<T, VisualStudioVersionMetadata>> items)
+        {
+            var best = items.FirstOrDefault();
+            foreach (var item in items.Skip(1))
+            {
+                if ((int)item.Metadata.Version > (int)best.Metadata.Version)
+                {
+                    best = item;
+                }
+            }
+
+            return best.Value;
```

Comment:
you access Value, will which die if this null.  Since, presumably this can't ever be null, you can use '.First()' above instead.